inc/server/class-form-server.php

```diff
@@ -176,6 +179,36 @@ public function register_routes() {
 				),
 			)
 		);
+		register_rest_route(
+			$namespace,
+			'/form/confirm',
+			array(
+				array(
+					'methods'             => WP_REST_Server::READABLE,
+					'callback'            => array( $this, 'confirm_form' ),
+					'permission_callback' => function ( $request ) {
+						$session = $request->get_param( 'session' );
+
+						if ( apply_filters( 'otter_form_session_confirmation', $session ) ) {
```

Comment:
With this step, we can verify the session id. The session can be a Stripe ID or something custom.